### PR TITLE
Fixed a bug where insertLineAbove would leave extra whitespace.

### PR DESCRIPTION
**Pull request by half-potato**
After some more testing by myself, I noticed a case where VSCode would leave extra whitespace. I have fixed this bug and added a unit test for this case.

### CODE REVIEW

File: src/actions/commands/actions.ts [hunk 1]
```diff
@@ -1536,28 +1536,34 @@ export class CommandInsertNewLineAbove extends BaseCommand {
     }
 
     vimState.cursors = getCursorsAfterSync(vimState.editor);
-    const indentAmt = Math.max(charPos - vimState.cursors[0].start.character, 0);
+    const endPos = vimState.cursors[0].start.character;
+    const indentAmt = charPos - endPos;
+    console.log(charPos);
+    console.log(indentAmt);
```
Comment (J-Fields) [inline src/actions/commands/actions.ts:1542]:
```suggestion
```

**Review by J-Fields — APPROVED** [review]
Thank you!
